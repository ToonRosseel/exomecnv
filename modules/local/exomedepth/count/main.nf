process COUNT {
    tag "$meta.id $meta2.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/r-exomedepth:1.1.16--r43hfb3cda0_3' :
        'biocontainers/r-exomedepth:1.1.16--r43hfb3cda0_3' }"

    publishDir "$params.outdir/exomedepth/counts", mode: 'copy'

    input:
    tuple val(meta), path(bam), path(bai)
    tuple val(meta2), path(exon_target)

    output:
    tuple val(meta), path("*.txt"), emit: counts
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script: // This script is bundled with the pipeline, in nf-cmgg/exomecnv/bin/

    def VERSION = '1.1.16'

    """
    ExomeDepth_count.R \\
        $meta.id \\
        $bam \\
        $bai \\
        $exon_target \\
        $meta2.id

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ExomeDepth: ${VERSION}
        R: \$(Rscript --version | sed 's/R scripting front-end //g')
    END_VERSIONS

    """

    stub:
    def VERSION = '1.1.16'
    """
    touch ${meta.id}_${meta2.id}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ExomeDepth: ${VERSION}
        R: \$(Rscript --version | sed 's/R scripting front-end //g')
    END_VERSIONS
    """
}
