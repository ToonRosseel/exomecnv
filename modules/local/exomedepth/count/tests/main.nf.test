nextflow_process {

    name "Test Process COUNT"
    script "../main.nf"
    process "COUNT"

    test("homo_sapiens - bam, bai, roi") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [ id:'exon' ],
                    file(params.roi_auto)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("homo_sapiens - bam, bai, roi - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [ id:'exon' ],
                    file(params.roi_auto)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
