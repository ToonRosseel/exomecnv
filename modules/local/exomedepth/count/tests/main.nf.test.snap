{
    "homo_sapiens - bam, bai, roi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_exon.txt:md5,38b5cbbe36f1561d05e9318c972f947a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aa3132f64e2e05283f1e8d6113ccaa9e"
                ],
                "counts": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_exon.txt:md5,38b5cbbe36f1561d05e9318c972f947a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa3132f64e2e05283f1e8d6113ccaa9e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T12:56:52.764127749"
    },
    "homo_sapiens - bam, bai, roi - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_exon.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aa3132f64e2e05283f1e8d6113ccaa9e"
                ],
                "counts": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_exon.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa3132f64e2e05283f1e8d6113ccaa9e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T13:00:02.923784879"
    }
}